### DISCUSSION TITLE
使用proxysql代理doris，无法判断fe的只读状态

### DISCUSSION THREAD

**Discussion by yidazhi** [apache/doris#33551] [A - General / Q&A]
版本：2.0.2
FE的日志如下：
2024-04-11 17:59:06,802 WARN (mysql-nio-pool-60|1626) [StmtExecutor.executeByLegacy():776] execute Exception. stmt[416348, b8f417c95beb4c8e-89a3952141028a53]
org.apache.doris.common.AnalysisException: errCode = 2, detailMessage = Unknown system variable 'read_only'

查看proxysql的mysql_server_read_only_log表
+--------------+------+------------------+-----------------+-----------+------------------------------------------------------------------+
| hostname     | port | time_start_us    | success_time_us | read_only | error                                                            |
+--------------+------+------------------+-----------------+-----------+------------------------------------------------------------------+
| 192.168.82.5 | 9030 | 1712824995284375 | 0               | NULL      | errCode = 2, detailMessage = Unknown system variable 'read_only' |
| 192.168.82.4 | 9030 | 1712824995298755 | 0               | NULL      | errCode = 2, detailMessage = Unknown system variable 'read_only' |
| 192.168.82.3 | 9030 | 1712824995313541 | 0               | NULL      | errCode = 2, detailMessage = Unknown system variable 'read_only' |

doris的monitor用户权限
      UserIdentity: 'default_cluster:monitor'@'%'
          Password: Yes
             Roles: 
       GlobalPrivs: Admin_priv 
      CatalogPrivs: NULL
     DatabasePrivs: internal.default_cluster:information_schema: Select_priv ; internal.default_cluster:mysql: Select_priv 
        TablePrivs: NULL
          ColPrivs: NULL
     ResourcePrivs: NULL
WorkloadGroupPrivs: normal: Usage_priv 

请教各位大佬，怎么才能让doris识别read_only

**Comment by yagagagaga** [comment]
Doris 目前没有这个系统变量，不过会在下一个版本中增加。
